WordPress/gutenberg
# Divider: Improve default behavior when in vertical orientation

Follow-up to #38842

## What?

- Conditionally renders the divider as an `inline` element when the `orientation` is `vertical`.
- Improves stories and documentation.

## Why?

This ensures that the divider is visible "as expected" in both orientations when inside a flow layout container (not flex). Initially I was only going to improve the docs to demonstrate proper usage in the flex layout case, but I realized that we can also add a conditional `display` declaration so it just works out of the box.

I'm inclined to classify this as a non-breaking change, because any proper usage of the vertical divider should already be `display: inline` or inside a flex container (in which case the `inline` won't matter).

## Testing Instructions

`npm run storybook:dev` and check out the stories for `Divider`.

- The `vertical` orientation works without a flex container.
- Inside a flex container, it still works as expected as long as it's `stretch` aligned.
